cli/kafka/config.go

```diff
@@ -44,18 +54,73 @@ type ConfigForIDK struct {
 	BatchMaxStaleness time.Duration
 	Timeout           time.Duration
 
-	Table   string
-	IDField string
-	Fields  []idk.RawField
+	Table       string
+	IDField     string
+	PrimaryKeys []string
+	Fields      []idk.RawField
+
+	Encode string
+}
+
+// Generate a Config struct based on a configuration file
+// Default values for Config struct are defined here
+func ConfigStructFromFile(cfgFile string) (cfg Config, err error) {
+	// configure viper
+	v := viper.New()
+	v.SetConfigFile(cfgFile)
+	v.SetConfigType("toml")
+
+	// set defaults
+	v.SetDefault("hosts", []string{"localhost:9092"})
+	v.SetDefault("group", "default-featurebase-group")
+	v.SetDefault("batch-size", 1)
+	v.SetDefault("batch-max-staleness", 5*time.Second)
+	v.SetDefault("timeout", 5*time.Second)
+	v.SetDefault("encode", JSON)
+
+	// Read the kafka config file.
+	err = v.ReadInConfig()
+	if err != nil {
+		return cfg, fmt.Errorf("error reading configuration file '%s': %v", cfgFile, err)
+	}
+
+	if err := v.Unmarshal(&cfg); err != nil {
+		return cfg, errors.Wrap(err, "unmarshalling config")
+	}
+
+	return
+
 }
 
 // ValidateConfig validates the config is usable.
+// Note that different encoding methods require
+// different configurations
 func ValidateConfig(c Config) error {
+
+	// validate common
 	if c.Table == "" {
 		return errors.Errorf("table is required")
-	} else if len(c.Topics) == 0 {
+	}
+
+	if len(c.Topics) == 0 {
 		return errors.Errorf("at least one topic is required")
-	} else if len(c.Fields) > 0 {
+	}
+
+	// validate on a by encoding basis
+	switch c.Encode {
```

Comment:
is `c.Encode` always expected to have a value? Can it be blank? In other words, do we need a default case here which returns an error? Or is it ok to leave all other cases unvalidated?

Reply:
Default value is define by viper and is JSON. Could add something here so it's clearer when reading though.